resources/session04/mashup.py

```diff
@@ -0,0 +1,213 @@
+    "high": "High Score"}
+
+def get_inspection_page(**kwargs):
+    url = INSPECTION_DOMAIN + INSPECTION_PATH
+    params = INSPECTION_PARAMS.copy()
+    for key, val in kwargs.items():
+        if key in INSPECTION_PARAMS:
+            params[key] = val
+    resp = requests.get(url, params=params)
+    resp.raise_for_status()
+    return resp.text
+
+
+def parse_source(html):
+    parsed = BeautifulSoup(html)
+    return parsed
+
+
+def load_inspection_page(name):
+    file_path = pathlib.Path(name)
+    with file_path.open('r') as fh:
+        return fh.read()
+
+
+def restaurant_data_generator(html):
+    id_finder = re.compile(r'PR[\d]+~')
+    return html.find_all('div', id=id_finder)
+
+
+def has_two_tds(elem):
+    is_tr = elem.name == 'tr'
+    td_children = elem.find_all('td', recursive=False)
+    has_two = len(td_children) == 2
+    return is_tr and has_two
+
+
+def clean_data(td):
+    return td.text.strip(" \n:-")
+
+
+def extract_restaurant_metadata(elem):
+    restaurant_data_rows = elem.find('table').find_all(
+        has_two_tds, recursive=False
+    )
+    rdata = {}
+    current_label = ''
+    for data_row in restaurant_data_rows:
+        key_cell, val_cell = data_row.find_all('td', recursive=False)
+        new_label = clean_data(key_cell)
+        current_label = new_label if new_label else current_label
+        rdata.setdefault(current_label, []).append(clean_data(val_cell))
+    return rdata
+
+
+def is_inspection_data_row(elem):
+    is_tr = elem.name == 'tr'
+    if not is_tr:
+        return False
+    td_children = elem.find_all('td', recursive=False)
+    has_four = len(td_children) == 4
+    this_text = clean_data(td_children[0]).lower()
+    contains_word = 'inspection' in this_text
+    does_not_start = not this_text.startswith('inspection')
+    return is_tr and has_four and contains_word and does_not_start
+
+
+def get_score_data(elem):
+    inspection_rows = elem.find_all(is_inspection_data_row)
+    samples = len(inspection_rows)
+    total = 0
+    high_score = 0
+    average = 0
+    for row in inspection_rows:
+        strval = clean_data(row.find_all('td')[2])
+        try:
+            intval = int(strval)
+        except (ValueError, TypeError):
+            samples -= 1
+        else:
+            total += intval
+            high_score = intval if intval > high_score else high_score
+
+    if samples:
+        average = total/float(samples)
+    data = {
+        u'Average Score': average,
+        u'High Score': high_score,
+        u'Total Inspections': samples
+    }
+    return data
+
+def add_color(inspection_data):
+    '''
+    Add color and symbols for high scores
+    '''
+    if inspection_data[u'Average Score'] > 50:
+        color = 'ff0000'
+    elif inspection_data[u'Average Score'] > 30:
+        color = 'ffff00'
+    else:
+        color = '33cc33'
+    if inspection_data[u'High Score'] > 80:
+        size = 'large'
+    else:
+        size = 'medium'
```

Comment:
by setting a default value of size and color at the top, you could get rid of both else clauses.  Might be a bit easier to see.